src/components/CurrentRefinedValues/CurrentRefinedValues.js

```diff
@@ -5,8 +5,13 @@ import Template from '../Template.js';
 import {isSpecialClick} from '../../lib/utils.js';
 import map from 'lodash/collection/map';
 import cloneDeep from 'lodash/lang/cloneDeep';
+import {isEqual} from 'lodash';
 
 class CurrentRefinedValues extends React.Component {
+  shouldComponentUpdate(nextProps) {
+    return !isEqual(this.props.refinements, nextProps.refinements);
```

Comment:
That would be awesome to only have to do so, currently not feasible without refactoring a lot I guess. I need to better understand how we could do so.